pytorch_lightning/core/hooks.py

```diff
@@ -138,11 +138,20 @@ def backward(self, use_amp, loss, optimizer):
                 else:
                     loss.backward()
 
+        .. note:: with PyTorch 1.6+ + precision=16 + multiple optimizers, set .backward(retrain_graph=True)
```

Comment:
The [example](https://pytorch.org/docs/master/notes/amp_examples.html#working-with-multiple-models-losses-and-optimizers) is misleading, I guess.  The `retain_graph=True` bit has nothing to do with Amp, it's only present because both losses interleave outputs from multiple models.  Both backward passes use the same model graphs, so the first backward() must not tear them down.  `retain_graph=True` would be necessary with or without Amp.  That's unclear and maybe I should either change the example snippet so `retain_graph=True` is not needed, or add a comment clarifying that `retain_graph=True` there is not Amp-related.